### PR TITLE
hmd preview

### PR DESCRIPTION
**Pull request by davidkelly**
When in HMD mode, the monitor can show either the camera's view or your HMD's view.  Whichever it is, that is what will show up in the preview window on the spectator camera's app in the tablet.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1422/HighFidelity-Beta-PR10908-b41c470064be7362a551d52dee68be4d79a5e139.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1422/HighFidelity-Beta-PR10908-b41c470064be7362a551d52dee68be4d79a5e139.exe

**Comment by davidkelly** [comment]
updated slightly, so the monitor will not use the hmdPreviewFrame texture, for now.  That way, though we see a stereo preview, the monitor is mono.  The hack needs to be removed once the texture is mono

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1431/HighFidelity-Beta-PR10908-93840bf93649ce2d81de4ed4732bee574ef09c34.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1431/HighFidelity-Beta-PR10908-93840bf93649ce2d81de4ed4732bee574ef09c34.exe

**Comment by zfox23** [comment]
[HMD preview in tablet app](https://trello.com/c/dDqYJX6S/171-hmd-preview-in-tablet-app)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1452/HighFidelity-Beta-PR10908-688b5dad8b9077407122e43b7580849fdc1f7ecb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1452/HighFidelity-Beta-PR10908-688b5dad8b9077407122e43b7580849fdc1f7ecb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1468/HighFidelity-Beta-PR10908-f18e64ed1aed5d3a7f765588a6d936619b1f31cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1468/HighFidelity-Beta-PR10908-f18e64ed1aed5d3a7f765588a6d936619b1f31cb.exe

**Comment by davidkelly** [comment]
Test Plan

In HMD, bring up spectator camera
with the Monitor Shows -> HMD Preview, the monitor should show what you are seeing in HMD, and the preview window should show the same.
with the Monitor Shows -> Camera View, the monitor should show what the camera sees (move it around), and the preview window should show same
In desktop, this need not be tested since we don't care, as this will not be a desktop feature.

File: libraries/model-networking/src/model-networking/TextureCache.cpp [hunk 1]
```diff
@@ -984,10 +984,30 @@ NetworkTexturePointer TextureCache::getResourceTexture(QUrl resourceTextureUrl)
             }
         }
     }
+    // FIXME: Generalize this, DRY up this code
+    if (resourceTextureUrl == HMD_PREVIEW_FRAME_URL) {
```
Comment (zfox23) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: this can be an `else if`, right?

Comment (davidkelly) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: Yea no reason not to.  Small gain -- one less string compare when falling through to return an empty texture.

Comment (davidkelly) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: But this is exactly the code that will get updated a bit when we generic-ize it.

Comment (zfox23) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: True dat.

**Comment by davidkelly** [comment]
NOTE:   we plan to make a more generic version of the code in TextureCache for the resource://<whatever> textures.  That way, one could add a new one without special member variables for the framebuffer and texture (say, maps with the url as the key, etc...).  However we want this in now, and don't want to make a generic mechanism for doing so yet (unless it is really desired).

**Comment by YuppyPlays** [comment]
Will test :)

Should I use previous version or wait for latest version to build? (as I see some checks haven't completed)

**Comment by davidkelly** [comment]
lets use latest for sure -- should have grab fixes, etc...

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1501/HighFidelity-Beta-PR10908-1c9c36a8c86792ffde459bc33719adadba96118b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1501/HighFidelity-Beta-PR10908-1c9c36a8c86792ffde459bc33719adadba96118b.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
In HMD, bring up spectator camera **Check**
With the Monitor Shows -> HMD Preview, the monitor should show what you are seeing in HMD, and the preview window should show the same. **Pass**
With the Monitor Shows -> Camera View, the monitor should show what the camera sees (move it around), and the preview window should show same **Pass**
In desktop, this need not be tested since we don't care, as this will not be a desktop feature. **Check**
🍻 **Check, will have a cold one to celebrate later :D**

**Pull request merged**